### DISCUSSION TITLE
rte.exec results do not reflect in finalHtml

### DISCUSSION THREAD

**Discussion by mihir-khandekar** [GrapesJS/grapesjs#4153] [Q&A]
Hello,

I am trying to insert ordered list into the current selected component using the following code.

```
const rte = editor.RichTextEditor.initRte(editor.getSelected().view.el);
rte.exec('insertOrderedList');
```

Although, this does work and the list gets created, when is extract the HTML using `editor.getHtml()`, the list is missing and instead I get a regular `<p>` tag in its place.

Even tried adding this `editor.getSelected().trigger('sync:content');` after the `rte.exec()` command but to no avail.

Also, is this also a valid option?

```
editor.Canvas.getDocument().execCommand('insertOrderedList');
```

Please help.

@artf  your guidance will be very helpful.

**Comment by artf** [comment]
Use the Component API instead of the RTE to update the selected component.

**Reply by mihir-khandekar** [reply]
the component API doesn't work either. on extracting HTML for components added through the component API, the order of these components in the HTML is messed up. What could be the reason?
